File: mambo-protocol/src/main/scala/org/mambo/protocol/messages/game/basic/SequenceNumberMessage.scala





















// Generated on 05/03/2013 10:34:36
// Scala version by Blackrush
package org.mambo.protocol.messages

import org.mambo.core.io._
import org.mambo.protocol.{Message => BaseMessage, _}
import org.mambo.protocol.types._
import org.mambo.protocol.enums._

trait SequenceNumberMessage extends MessageDeserializer {
  val messageId = 6317

  protected trait Message extends BaseMessage with Serializable {
    val messageId = SequenceNumberMessage.messageId

    val number: UShort

    def serialize(buf: Buffer) {
      buf.writeUShort(number)
    }

  }

  protected def create(number: UShort): Target = throw new IllegalAccessException

  def deserialize(buf: Buffer) = {
    val number = buf.readUShort()

    create(number)
  }

}

object SequenceNumberMessage extends SequenceNumberMessage {
  case class MessageImpl(number: UShort) extends Message
  type Target = MessageImpl

  override def create(number: UShort) = MessageImpl(number)

  def apply(number: UShort) = create(number)
  def unapply(msg: MessageImpl) = MessageImpl.unapply(msg)
}

File: mambo-protocol/src/main/java/org/mambo/protocol/messages/SequenceNumberMessage.java


// Generated on 05/08/2013 19:37:41
package org.mambo.protocol.messages;

import java.util.*;
import org.mambo.protocol.types.*;
import org.mambo.protocol.enums.*;
import org.mambo.protocol.*;
import org.mambo.core.io.*;

public class SequenceNumberMessage extends NetworkMessage {
    public static final int MESSAGE_ID = 6317;
    
    @Override
    public int getNetworkMessageId() {
        return MESSAGE_ID;
    }
    
    public int number;
    
    public SequenceNumberMessage() { }
    
    public SequenceNumberMessage(int number) {
        this.number = number;
    }
    
    @Override
    public void serialize(Buffer buf) {
        buf.writeUShort(number);
    }
    
    @Override
    public void deserialize(Buffer buf) {
        number = buf.readUShort();
        if (number < 0 || number > 65535)
            throw new RuntimeException("Forbidden value on number = " + number + ", it doesn't respect the following condition : number < 0 || number > 65535");
    }
    
}
